include/pybind11/cast.h

```diff
@@ -2343,8 +2382,8 @@ constexpr bool args_are_all_positional() {
 template <return_value_policy policy,
           typename... Args,
           typename = enable_if_t<args_are_all_positional<Args...>()>>
```

Comment:
Connected to the capacity idea above: `enable_if` only if <= capacity, is that feasible?

I think we don't have to be stingy with the capacity, e.g. 128 (pointers) would seem totally fine, so in practice this would hardly ever go to the fallback.